### PR TITLE
Re-activate API tests

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp #10112

.. Oups .. I accidentally disabled the API tests

And when re-enabling them, I found some really ugly bugs as well ..

Like accidentally commiting transactions because of the `autocommit` flag inside the `DBA::lock()` methods. Because this setting overrules the transaction hierarchy for tests and commits the current test-content straight forward into the DB ...

Sadly calculating the codecov-reporting xml now takes a lot longer than before, so we're back at ~4 minutes test-duration :-/

### CODE REVIEW

File: tests/legacy/ApiTest.php [hunk 1]
```diff
@@ -2308,12 +2309,13 @@ public function testApiConvertItemWithoutBody()
...
 				'title'   => 'item_title',
+				'uri-id'  => 1,
 				'body'    => '',
 				'plink'   => 'item_plink'
 			]
 		);
-		self::assertEquals('item_title', $result['text']);
-		self::assertEquals('<h4>item_title</h4><br>item_plink', $result['html']);
+		self::assertEquals("item_title\n\nParent status", $result['text']);
+		self::assertEquals('<h4>item_title</h4><br>Parent statusitem_plink', $result['html']);
```
Comment (nupplaphil) [inline tests/legacy/ApiTest.php]: @annando - Are these changes really intended? I had to adapt these assertions because the output of `api_convert_item()` changed that way

**Review by MrPetovan — APPROVED** [review]

File: tests/legacy/ApiTest.php (hunk 1 above)
Comment (annando) [reply tests/legacy/ApiTest.php]: If possible, we now store media as attached data. In the API this is then added to the body again. This is why you now have to provide an `uri-id`. And depending upon the value of the `uri-id` different attachments are added. If you don't want to have added something, you should use an `uri-id` that hasn't been used before.

Comment (nupplaphil) [reply tests/legacy/ApiTest.php]: I see, thanks!

**Comment by nupplaphil** [comment]
I adjusted the test, so it's ready now, @annando thanks for the hint

**Review by MrPetovan — APPROVED** [review]
